yarn-project/rollup-provider/src/app.ts

```diff
@@ -269,6 +269,23 @@ export function appFactory(node: AztecNode, prefix: string) {
     ctx.status = 200;
   });
 
+  router.post('/tx-simulate', checkReady, async (ctx: Koa.Context) => {
+    const stream = new PromiseReadable(ctx.req);
+    const postData = (await stream.readAll()) as Buffer;
+    const tx = Tx.fromBuffer(postData);
+    try {
+      await node.simulatePublicPart(tx);
+    } catch (err) {
+      if (err instanceof SimulationError) {
+        ctx.body = {
+          simulationError: err.toJSON(),
+        };
```

Comment:
Do we need to also set the http code to 4xx here, given this is not a JSON RPC interface but a sort-of RESTful one? (with emphasis on the sort-of)